src/Core/Usecase/User/ResetUserPassword.php

```diff
@@ -62,22 +64,28 @@ public function setRepository(ResetPasswordRepository $repo)
 	public function interact()
 	{
 		$token = urldecode($this->getPayload('token'));
-		// Check if the reset token is valid
-		if ($this->repo->isValidResetToken($token)) {
-			// If its valid..
-			// Change the password
-			$password = $this->getPayload('password');
-			$this->repo->setPassword($token, $password);
+		$password = $this->getPayload('password');
+		$entity_array = [
+			'token' => $token,
+			'password' => $password
+		];
 
-			// And delete the token
-			$this->repo->deleteResetToken($token);
-		} else {
-			// Otherwise throw an exception
-			throw new ValidatorException('Invalid or expired token', ['Invalid or expired reset token']);
-		}
+		$this->verifyValid($entity_array);
+			
+		$this->repo->setPassword($token, $password);
+
+		// And delete the token
+		$this->repo->deleteResetToken($token);
```

Comment:
I'm not sure how much of an issue it is but may be we should delete the token before we save the password, or does it matter?